### ISSUE TITLE
Strange Bug involving async

### ISSUE THREAD

**Issue by sunmy2019** [microsoft/pyright#8232]
Sorry that I cannot come up with a better title.

**Describe the bug**
I got a strange error in my project. I tried to minimize it to the following code pieces.

**Code or Screenshots**

```python
import asyncio


async def foo(xs: list[list[int]]):
    for x in xs:

        async def find_first():
            for y in x:
                return y
            return None

        answer: list = []

        print(await find_first())

        if answer:
            pass


asyncio.run(foo([[1, 2, 3], [4, 5, 6]]))

```

Run it with python:
```
1
4
```

run it with pyright cli
```
a.py
  a.py:14:21 - error: "int" is not awaitable
    "int" is incompatible with protocol "Awaitable[_T_co@Awaitable]"
      "__await__" is not present (reportGeneralTypeIssues)
  a.py:14:21 - error: "None" is not awaitable
    "None" is incompatible with protocol "Awaitable[_T_co@Awaitable]"
      "__await__" is not present (reportGeneralTypeIssues)
2 errors, 0 warnings, 0 informations
```

version
```
pyright 1.1.369
```

**Comment by wch** [comment]
This looks very similar to #8229.

**Comment by erictraut** [comment]
Yes, this is the same bug as #8229. I'll leave both open for now and mark both fixed when I have a solution in place.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.370
